examples/api/lib/widgets/inherited_model/inherited_model.0.dart

```diff
@@ -0,0 +1,169 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for InheritedModel
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const InheritedModelApp());
+
+class InheritedModelApp extends StatelessWidget {
+  const InheritedModelApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: InheritedModelExample(),
+    );
+  }
+}
+
+class FooBarModel extends InheritedModel<String> {
+  const FooBarModel({
+    super.key,
+    required this.fooSize,
+    required this.barSize,
+    required super.child,
+  });
+
+  final double fooSize;
+  final double barSize;
+
+  static FooBarModel? of(BuildContext context, String aspect) {
+    return InheritedModel.inheritFrom<FooBarModel>(context, aspect: aspect);
+  }
+
+  @override
+  bool updateShouldNotify(FooBarModel oldWidget) {
+    return fooSize != oldWidget.fooSize
+     || barSize != oldWidget.barSize;
+  }
+
+  @override
+  bool updateShouldNotifyDependent(FooBarModel oldWidget, Set<String> dependencies) {
+    if (fooSize != oldWidget.fooSize && dependencies.contains('foo')) {
+      print('Rebuilding Foo Widget');
```

Comment:
Having these prints is a little strange.... can we just show in the UI what got rebuilt?